tests/strategies/settlement/test_settlement_strategy.py

```diff
@@ -0,0 +1,120 @@
+        self.market_mock.id = str(uuid.uuid4())
+        self.test_bid = Bid("123", self.time_slot, 10, 1, buyer="test_name")
+        self.test_offer = Offer("234", self.time_slot, 50, 1, seller="test_name")
+        self.market_mock.bid = MagicMock(return_value=self.test_bid)
+        self.market_mock.offer = MagicMock(return_value=self.test_offer)
+        self.market_mock.bids = {self.time_slot: self.test_bid}
+        self.area_mock = Mock()
+        self.area_mock.name = "test_name"
+        self.area_mock.uuid = str(uuid.uuid4())
+        self.settlement_markets = {
+            self.time_slot: self.market_mock
+        }
+
+    def _setup_strategy_fixture(self, strategy_fixture, should_post_bid, should_post_offer):
+        strategy_fixture.owner = self.area_mock
+        strategy_fixture.state.set_energy_measurement_kWh(1, self.time_slot)
+        strategy_fixture.state.should_post_bid = MagicMock(return_value=should_post_bid)
+        strategy_fixture.state.should_post_offer = MagicMock(return_value=should_post_offer)
+        strategy_fixture.area = Mock()
+        strategy_fixture.area.settlement_markets = self.settlement_markets
+        strategy_fixture.get_market_from_id = MagicMock(return_value=self.market_mock)
+
+    def teardown_method(self):
+        ConstSettings.GeneralSettings.ENABLE_SETTLEMENT_MARKETS = False
+
+    @pytest.mark.parametrize(
+        "strategy_fixture", [LoadHoursStrategy(100), PVStrategy()])
+    @pytest.mark.parametrize("should_post_bid", [True, False])
+    @pytest.mark.parametrize("should_post_offer", [True, False])
+    def test_event_market_cycle_posts_bids_and_offers(
+            self, strategy_fixture, should_post_bid, should_post_offer):
+        self._setup_strategy_fixture(strategy_fixture, should_post_bid, should_post_offer)
+        self.settlement_strategy.event_market_cycle(strategy_fixture)
+        if should_post_bid:
+            self.market_mock.bid.assert_called_once_with(
+                10.0, 1.0, self.area_mock.name, original_bid_price=10.0,
+                buyer_origin=self.area_mock.name, buyer_origin_id=self.area_mock.uuid,
+                buyer_id=self.area_mock.uuid, attributes=None, requirements=None
+            )
+        if should_post_offer:
+            self.market_mock.offer.assert_called_once_with(
+                price=50.0, energy=1.0, seller=self.area_mock.name,
+                seller_origin=self.area_mock.name,
+                seller_origin_id=self.area_mock.uuid, seller_id=self.area_mock.uuid
+            )
+
+    @pytest.mark.parametrize(
+        "strategy_fixture", [LoadHoursStrategy(100), PVStrategy()])
+    @pytest.mark.parametrize("should_post_bid", [True, False])
+    @pytest.mark.parametrize("should_post_offer", [True, False])
+    def test_event_tick_updates_bids_and_offers(
+            self, strategy_fixture, should_post_bid, should_post_offer):
+        self._setup_strategy_fixture(strategy_fixture, should_post_bid, should_post_offer)
```

Comment:
Interesting way of looping over different cases! Will keep this in mind for future. Thanks!

Reply:
Kudos to @fievelk that pointed out some PRs ago that pytest.mark.parametrize applies cartesian product to arguments that are added one after the other!